lib/sww.js

```diff
@@ -217,17 +223,42 @@ ServiceWorkerWare.normalizeMwAnswer = function (answer, request, response) {
 };
 
 /**
- * Walk all the middle ware installed asking if they have prerequisites
- * (on the way of a promise to be resolved) when installing the SW
+ * Walk all the middleware installed asking if they have prerequisites
+ * (on the way of a promise to be resolved) when installing the SW.
  */
 ServiceWorkerWare.prototype.onInstall = function sww_oninstall(evt) {
-  var waitingList = [];
-  this.middleware.forEach(function(mw) {
-    if (typeof mw.onInstall !== 'undefined') {
-      waitingList.push(mw.onInstall());
+  var installation = this.getFromMiddleware('onInstall');
+  evt.waitUntil(installation);
+};
+
+/**
+ * Walk all the installed middleware asking if they have prerequisites
+ * (on the way of a promise to be resolved) when SW activates.
+ */
+ServiceWorkerWare.prototype.onActivate = function sww_oninstall(evt) {
```

Comment:
Sure. I don't know why its not there. I will fix the name for the function as well.